Plugins/GitUIPluginInterfaces/ManagedExtensibility.cs

```diff
@@ -79,6 +86,21 @@ private static ExportProvider CreateExportProvider(string applicationDataFolder)
             return exportProviderFactory.CreateExportProvider();
         }
 
+        private static IEnumerable<FileInfo> GetPluginFiles(params string[] pluginsPaths)
+        {
+            var result = Enumerable.Empty<FileInfo>();
+
+            foreach (string pluginsPath in pluginsPaths)
+            {
+                if (!string.IsNullOrEmpty(pluginsPath) && Directory.Exists(pluginsPath))
+                {
+                    result = Enumerable.Concat(result, new DirectoryInfo(pluginsPath).GetFiles("*.dll", SearchOption.AllDirectories));
+                }
+            }
+
+            return result;
+        }
```

Comment:
1. Tests?
2. Do we need to load _all_ \*.dlls or only those matching a specific patten (e.g. GitExtensions.*.dll)? 
3. Is there a perf risk of searching for a looooooong time? 🤔

Reply:
1. Yeah, tests. I'm sorry, I'm going to add them.
2. I had a discussion with @mast-eu that GE org could reserve prefix GitExtensions.* on nuget.org, so that official/core plugins could be in this form and community plugins will have any other pattern.
3. The idea is that there will only be 1 level of folder nesting.

Reply:
2. The current plugins have no common naming scheme. So, at least for the moment, we need to load all DLLs from `defaultPluginPath`. A decision about enforcing a naming scheme or not can be taken later.
3. No, I don't expect a performance impact. I didn't perform any tests yet though.

Reply:
2. The current plugins have no common naming scheme. So, at least for the moment, we need to load all DLLs from `defaultPluginPath`. A decision about enforcing a naming scheme or not can be taken later.
3. No, I don't expect a performance impact. I didn't perform any tests yet though.

Reply:
> GE org could reserve prefix GitExtensions.* on nuget.org

👍  Any hints how to do that?


> 3\. The idea is that there will only be 1 level of folder nesting.

I'm afraid we can't assume that. We need to be strict in our implementation and don't go deeper than one level.

Reply:
> > GE org could reserve prefix GitExtensions.* on nuget.org
> 
> 👍 Any hints how to do that?

Send email to account@nuget.org, documentation https://docs.microsoft.com/en-us/nuget/reference/id-prefix-reservation#id-prefix-reservation-application-process.

> 
> > 3. The idea is that there will only be 1 level of folder nesting.
> 
> I'm afraid we can't assume that. We need to be strict in our implementation and don't go deeper than one level.

Ok than.